src/access_nri_intake/source/utils.py

```diff
@@ -15,6 +17,80 @@ class EmptyFileError(Exception):
     pass
 
 
+@dataclass
+class _AccessNCFileInfo:
+    """
+    Holds information about a NetCDF file that is used to create an intake-esm
+    catalog entry.
+
+    ______
+    Notes:
+    Use of both path and filename seems redundant, but constructing filename from
+    the path using a __post_init__ method makes testing more difficult. On balance,
+    more explicit tests are probably more important than the slight redundancy.
```

Comment:
I don't mind a bit of redundancy here - it's a negligible additional thing we need to track that makes life a bit easier (and then only at catalog build time, which is already a ~1hr HPC job).